client/hotrod/src/main/java/org/infinispan/hotrod/HotRodMutinyCaches.java

```diff
@@ -57,6 +60,19 @@ public Uni<Void> removeTemplate(String name) {
 
    @Override
    public Multi<String> templateNames() {
-      return null;
+      return Multi.createFrom().deferred(() -> toMulti(hotrod.transport.getTemplateNames()));
+   }
+
+   private Multi<String> toMulti(CompletionStage<Set<String>> cs) {
```

Comment:
This can just be `return Multi.createFrom().completionStage(cs).onItem().transformToIterable(Function.identity());`